docs/source/04_user_guide/06_pipelines.md

```diff
@@ -533,6 +533,52 @@ kedro run --runner=ParallelRunner
 
 > *Note:* You cannot use both `--parallel` and `--runner` flags at the same time (e.g. `kedro run --parallel --runner=SequentialRunner` raises an exception).
 
+#### Using a custom runner
+
+If the built-in runners don't meet your requirements, you can define your own runner in your project instead. For example, you may want to add a dry runner, a runner which lists which nodes would be run instead of executing them. You can define it in the following way (courtesy of [AntonyMilneQB](https://github.com/AntonyMilneQB])):
+
+```python
+# in your_project/src/your_project/runner.py
+from kedro.io import AbstractDataSet, DataCatalog, MemoryDataSet
+from kedro.pipeline import Pipeline
+from kedro.runner.runner import AbstractRunner
+
+
+class DryRunner(AbstractRunner):
+    """``DryRunner`` is an ``AbstractRunner`` implementation. It can be used to list which
+    nodes would be run without actually executing anything.
+    """
```

Comment:
Happy for you to ignore, but do you think providing an example would be helpful here (e.g to show what the output would look like)?

Reply:
I think this is a good idea to include somewhere. FYI the test output I wrote before looks like this:
```
test_sequential_runner.py F2020-03-18 14:25:30,420 - kedro.runner.dry_runner - INFO - Actual run would execute 5 nodes:
node5: random(None) -> [A]
node3: identity([A]) -> [B]
node4: identity([B]) -> [C]
node2: identity([C]) -> [D]
node1: identity([D]) -> [E]
2020-03-18 14:25:30,421 - kedro.runner.dry_runner - INFO - Pipeline execution completed successfully.
2020-03-18 14:25:30,422 - kedro.io.data_catalog - INFO - Loading data from `E` (MemoryDataSet)...
```

Reply:
Added